binding.gyp

```diff
@@ -4,9 +4,14 @@
 #
 
 {
+    'conditions': [
+        ['OS=="mac"', {'variables': {'NODE_OS_NAME': 'darwin'}},
+         'OS=="win"', {'variables': {'NODE_OS_NAME': 'win32'}},
+         {'variables': {'NODE_OS_NAME': '<(OS)'}}],
```

Comment:
This seems reasonable based on [the docs](https://gyp.gsrc.io/docs/InputFormatReference.md#predefined-variables), but I could see failing if `OS` is not `win`, `mac`, or `linux`. For example, if it's unset or `freebsd`, we probably want to show some error.

Feel free to ignore—not sure this is a good suggestion.

Reply:
The last line defaults to passing through gyp's OS as the NODE_OS_NAME. That happens to be [correct](https://nodejs.org/api/os.html#os_os_platform) for `freebsd` and `openbsd`, and we can handle more unusual platforms as they come.

Reply:
Ah, the gyp docs are really bad, but their test cases include the BSDs, and the setting *is* populated automatically when run by node-gyp.

P.S. I cringe every time I write this tool's name…